src/allmydata/node.py

```diff
@@ -265,7 +364,9 @@ def __init__(self, config, basedir=u"."):
 
     def init_tempdir(self):
         tempdir_config = self.config.get_config("node", "tempdir", "tmp").decode('utf-8')
-        tempdir = abspath_expanduser_unicode(tempdir_config, base=self.basedir)
+        tempdir = self.config.get_config_path(tempdir_config)
+        tempdir0 = abspath_expanduser_unicode(tempdir_config, base=self.config._basedir)
```

Comment:
Another use of the probably-shouldn't-be-used-from-here `_Config._basedir` here.

It seems like the use of `abspath_expanduser_unicode` here is only to be able to make the assertion on the next line that nothing has changed - or put another way, that `tempdir_config` was already an absolute path.  Maybe there's a way to make this assertion without using `self.config._basedir`?  If the path is absolute, after all, does it matter what path it is expanded relative to?  And there's `os.path.isabs` which would just answer the question without trying any expansion (but which doesn't do the extra Windows and unicode-y stuff that `abspath_expanduser_unicode` does...).

Unfortunately it looks like this is a really hairy operation and there are probably a ton of ways to get it wrong... :(

A simpler fix would be to put this logic onto `_Config` and then just invoke it from here, keeping everything the same but avoiding the private attribute access.

Reply:
I've removed `tempdir0` -- that was just debugging so I was sure the "get_config_path" was doing the right stuff here